incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -15,6 +15,201 @@ import (
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 )
 
+var _ = Describe("Exception", func() {
+	ctx := context.Background()
+
+	BeforeEach(func() {
+		// We want to ensure we're working with a clean slate, in case a previous tests objects still exist
+		cleanupObjects(ctx)
+	})
+
+	AfterEach(func() {
+		// Change current config back to the default value.
+		resetHNCConfigToDefault(ctx)
+		cleanupObjects(ctx)
+	})
+
+	Context("Add exception annotations ", func() {
+		const (
+			p  = "parent"
+			c1 = "child1"
+			c2 = "child2"
+			c3 = "child3"
+		)
+		tests := []struct {
+			name         string
+			selector     string
+			treeSelector string
+			want         []string
+			notWant      []string
+		}{
+			{name: "simply using select",
+				selector:     "!" + c1 + api.LabelTreeDepthSuffix,
+				treeSelector: "",
```

Comment:
You can omit anything with the zero value (for a string, this is `""`)

Reply:
Good to know! Would you prefer `0` over `""`?

Reply:
No, I just mean: take out this line completely, and Go will auto-set `treeSelector` to `""`.

The term "zero value" just means "default value" in Go. For an `int`, the zero value is `0`; for a string, it's `""`, for a pointer it's `nil`, etc.

Reply:
I see! Done